e2e/src/test/java/uk/gov/pay/java_lambdas/integration/bin_ranges/BinRangesIngestIT.java

```diff
@@ -0,0 +1,296 @@
+            .bucket(PROMOTED_BUCKET_NAME).build()).contents().size());
+        assertEquals(0, s3Client.listObjects(ListObjectsRequest.builder()
+            .bucket(STAGED_BUCKET_NAME).build()).contents().size());
+        logger.debug("invoking bin-ranges-transfer");
+        var transferFunctionInvokeRes = lambdaClient.invoke(InvokeRequest.builder()
+            .functionName("bin-ranges-transfer")
+            .build()
+        );
+
+        JSONObject transferFunctionResponseCandidate = commonAssertions(transferFunctionInvokeRes.payload());
+        SdkBytes diffFunctionRequestPayload = SdkBytes.fromUtf8String(transferFunctionResponseCandidate.toString());
+
+        logger.debug("invoking bin-ranges-diff");
+        var diffFunctionInvokeRes = lambdaClient.invoke(InvokeRequest.builder()
+            .functionName("bin-ranges-diff")
+            .payload(diffFunctionRequestPayload)
+            .build()
+        );
+
+        JSONObject diffFunctionResponseCandidate = commonAssertions(diffFunctionInvokeRes.payload());
+        SdkBytes integrityFunctionRequestPayload = SdkBytes.fromUtf8String(diffFunctionResponseCandidate.toString());
+
+        logger.debug("invoking bin-ranges-integrity");
+        var integrityFunctionInvokeRes = lambdaClient.invoke(InvokeRequest.builder()
+            .functionName("bin-ranges-integrity")
+            .payload(integrityFunctionRequestPayload)
+            .build()
+        );
+
+        JSONObject integrityFunctionResponseCandidate = commonAssertions(integrityFunctionInvokeRes.payload());
+        SdkBytes promotionFunctionRequestPayload = SdkBytes.fromUtf8String(integrityFunctionResponseCandidate.toString());
+
+        logger.debug("invoking bin-ranges-promotion");
+        var promotionFunctionInvokeRes = lambdaClient.invoke(InvokeRequest.builder()
+            .functionName("bin-ranges-promotion")
+            .payload(promotionFunctionRequestPayload)
+            .build()
+        );
+
+        boolean promotionFunctionCandidate = Boolean.parseBoolean(new String(promotionFunctionInvokeRes.payload().asByteArray()));
+        assertTrue(promotionFunctionCandidate);
+        assertEquals(2, s3Client.listObjects(ListObjectsRequest.builder()
+            .bucket(PROMOTED_BUCKET_NAME).build()).contents().size());
+    }
+    
+    // --- PRIVATE ---
+
+    private JSONObject commonAssertions(SdkBytes responsePayload) {
+        JSONObject underTest = new JSONObject(new String(responsePayload.asByteArray()));
+        assertTrue((Boolean) underTest.get("proceed"));
+        assertTrue(((String) underTest.get("s3Key")).contains(V03_FILENAME));
+        return underTest;
+    }
+
+    private static String testcontainersLabels() {
+        return Stream.concat(
+                Stream
+                    .of(DockerClientFactory.DEFAULT_LABELS.entrySet().stream(),
+                        ResourceReaper.instance().getLabels().entrySet().stream())
+                    .flatMap(Function.identity())
```

Comment:
Would
```
Stream.concat(
  DockerClientFactory.DEFAULT_LABELS.entrySet().stream(),  
  ResourceReaper.instance().getLabels().entrySet().stream()
)
```
not work - rather than use two different ways of combining streams in one method?

Reply:
I'll simplify